druid/src/widget/widget_wrapper.rs

```diff
@@ -0,0 +1,47 @@
+/// A trait for widgets that wrap a single child to expose that child for access and mutation
+pub trait WidgetWrapper {
```

Comment:
Do we actually make use of the trait-ness of this trait anywhere?

My inclination would be to just have independent methods on each widget, instead of a trait (which needs to be in scope). That way we would have just `Container::child` and `Container::child_mut` methods.

An annoying consideration here is also that we are inconsistent about monomorphizing widget params; that is, some single-child containers are parameterized over their child type, and some just box unilaterally.

Reply:
If we don't have the trait, I need to implement them all over again in bindings in order to treat this generically, and so would any other scheme trying to do the same thing. I guess coherence sometimes forces things like that anyway, but it didn't yet.
The lack of traits in Contexts does cause real problems with duplicated code.

Also, these methods are fairly useless for anything that isn't some kind of general scheme for access. 
To use them in a dynamic way (which is the only real use case, otherwise you would set it up in construction), you need to implement so much boiler plate (controllers, commands, navigation, threading through widget ids), that an import seems like the last of our worries.

I have to admit I don't agree with the 'hanging methods off of traits' objection really. Importing is just normal, and IDEs will do it for you.